lib/logger/lib/logger/utils.ex

```diff
@@ -109,129 +114,57 @@ defmodule Logger.Utils do
+    args: [],
+    control_char: ?s,
+    encoding: :unicode,
+    pad_char: ?\s,
+    precision: :none,
+    strings: true,
+    width: :none
+  }
 
-  defp collect_cc(:encoding, [?t | t], args, used_format, used_args, opts),
-    do: collect_cc(:done, t, args, [?t | used_format], used_args, opts)
+  defp handle_format_spec(%{control_char: char} = spec, opts) when char in 'wWpP' do
+    %{args: args, width: width, strings: strings?} = spec
 
-  defp collect_cc(:encoding, t, args, used_format, used_args, opts),
-    do: collect_cc(:done, t, args, used_format, used_args, opts)
+    opts = %{
+      opts |
+      charlists: inspect_charlists(strings?, opts),
+      limit: inspect_limit(char, args, opts),
+      width: inspect_width(char, width),
+    }
 
-  ## done
-
-  defp collect_cc(:done, [?W | t], [data, limit | args], _used_format, _used_args, opts),
-    do: collect_inspect(t, args, data, %{opts | limit: limit, width: :infinity})
-
-  defp collect_cc(:done, [?w | t], [data | args], _used_format, _used_args, opts),
-    do: collect_inspect(t, args, data, %{opts | width: :infinity})
-
-  defp collect_cc(:done, [?P | t], [data, limit | args], _used_format, _used_args, opts),
-    do: collect_inspect(t, args, data, %{opts | limit: limit})
-
-  defp collect_cc(:done, [?p | t], [data | args], _used_format, _used_args, opts),
-    do: collect_inspect(t, args, data, opts)
-
-  defp collect_cc(:done, [h | t], args, used_format, used_args, _opts) do
-    {args, used_args} = collect_cc(h, args, used_args)
-    {t, args, [h | used_format], used_args}
-  end
-
-  defp collect_cc(?x, [a, prefix | args], used), do: {args, [prefix, a | used]}
-  defp collect_cc(?X, [a, prefix | args], used), do: {args, [prefix, a | used]}
-  defp collect_cc(?s, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?e, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?f, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?g, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?b, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?B, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?+, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?#, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?c, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?i, [a | args], used), do: {args, [a | used]}
-  defp collect_cc(?~, args, used), do: {args, used}
-  defp collect_cc(?n, args, used), do: {args, used}
-
-  defp collect_inspect(t, args, data, opts) do
-    data =
-      data
-      |> Inspect.Algebra.to_doc(opts)
-      |> Inspect.Algebra.format(opts.width)
-    {t, args, 'st~', [data]}
-  end
-
-  defp collect_value(current, [?* | t], [arg | args], used_format, used_args, opts, next)
-      when is_integer(arg) do
-    collect_cc(next, t, args, [?* | used_format], [arg | used_args],
-               put_value(opts, current, arg))
+    %{@inspected_format_spec | args: [inspect_data(args, opts)]}
```

Comment:
Question: shouldn't we change the original spec instead of modifying a default one?

Reply:
Well, my reason to do this way was to make the result of `scan_inspect` the most near possible to the old `inspect` one.

The old one was changing, for example, `"~w"` with `[:test]` args by `"~ts"` with `[":test"]` args 

So I just made a `:io_lib.scan_format('~ts', [:test])` and got:

```
[%{adjust: :right, args: [:test], control_char: 115, encoding: :unicode,
   pad_char: 32, precision: :none, strings: true, width: :none}]
```

This is what I saved to `@inspected_format_spec` (except the args key, which we change with the inspected args).

Now theorically, it will be a good idea to use the original spec, but I don't know if there are operators that just don't make sense or even fail for a `~s`.

Reply:
I see, that makes total sense to me then. Thank you!

Reply:
About my doubt, after some research I've reached to this docs:

- `control_char` is the type of control sequence: `$P`, `$w`, and so on. 
  - This we must replace by `$s`
- `args` is a list of the arguments used by the control sequence, or an empty list if the control sequence does not take any arguments.
  - This we must replace with `[inspected_data]`
- `width` is the field width.
  - This we bypass to `inspect` in the relevant cases, so I guess we can set to `:none` with
- `adjust` is the adjustment.
  - This can be a problem, since we totally ignore
- `precision` is the precision of the printed argument.
  - This can also be a problem, since we totally ignore
- `pad_char` is the padding character.
  - This can also be a problem, since we totally ignore
- `encoding` is set to true if translation modifier `t` is present.
  - In this one we force to `:unicode`, which means `true`
- `strings` is set to `false` if modifier `l` is present.
  - This we use when inspecting